File: app/src/main/java/com/example/crudapplication/presentation/activity/DetailActivity.java
package com.example.crudapplication.presentation.activity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.lifecycle.ViewModelProvider;

import com.example.crudapplication.R;
import com.example.crudapplication.databinding.ActivityDetailBinding;
import com.example.crudapplication.presentation.viewmodel.UserViewModel;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class DetailActivity extends AppCompatActivity {
    private ActivityDetailBinding binding; // View Binding 객체 선언
    private String uuidString; //id값 저장을 위한

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        initializeViews();
        getData();
        setupEditButton();
        setupCallButton();
        setupMessageButton();
    }

    // View Binding 초기화
    private void initializeViews(){
        binding = ActivityDetailBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
    }

    // MainActivity에서 전달한 데이터받아와서 뿌려줌
    private void getData(){
        Intent intent = getIntent();
        uuidString = intent.getStringExtra("uuid");
        binding.tvName.setText(intent.getStringExtra("name"));
        binding.tvPhoneValue.setText(intent.getStringExtra("phone"));
        binding.tvAddressValue.setText(intent.getStringExtra("address"));
    }


    // 편집버튼 눌렀을때
    private void setupEditButton() {
        Button editButton = findViewById(R.id.btn_edit);
        editButton.setOnClickListener(v -> {
            // EditUserActivity로 이동하면서 데이터 전달
            Intent intent = new Intent(DetailActivity.this, EditUserActivity.class);
            intent.putExtra("id", uuidString);  // id 값 전달
            intent.putExtra("name", binding.tvName.getText().toString());
            intent.putExtra("phone", binding.tvPhoneValue.getText().toString());
            intent.putExtra("address", binding.tvAddressValue.getText().toString());
//            Log.d("id", String.valueOf(userId));
//            Log.d("name", tvName.getText().toString());
//            Log.d("phone", tvPhone.getText().toString());
//            Log.d("address", tvAddress.getText().toString());
            startActivity(intent);
        });
    }

    // 전화걸기
    private void setupCallButton(){
        Button callButton = findViewById(R.id.btn_call);
        callButton.setOnClickListener(v -> {
            String phoneNumber = binding.tvPhoneValue.getText().toString();

            Intent callIntent = new Intent(Intent.ACTION_DIAL); // 전화 다이얼로그 화면으로 이동
            callIntent.setData(Uri.parse("tel:" + phoneNumber)); // phoneNumber 전달
            startActivity(callIntent);
        });
    }

    // 메시지 화면으로
    private void setupMessageButton(){
        Button messageButton = findViewById(R.id.btn_message);
        messageButton.setOnClickListener(v -> {
            String phoneNumber = binding.tvPhoneValue.getText().toString();

            Intent messageIntent = new Intent(Intent.ACTION_SENDTO); // 메시지 화면으로 이동
            messageIntent.setData(Uri.parse("smsto:" + phoneNumber)); // phoneNumber 전달
            startActivity(messageIntent);
        });
    }

}

File: app/src/main/java/com/example/crudapplication/presentation/activity/DetailActivity.kt
package com.example.crudapplication.presentation.activity

import android.content.Intent
import android.graphics.BitmapFactory
import android.net.Uri
import android.os.Bundle
import android.util.Base64
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.example.crudapplication.R
import com.example.crudapplication.databinding.ActivityDetailBinding
import com.google.android.material.floatingactionbutton.FloatingActionButton
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class DetailActivity : AppCompatActivity() {
    private lateinit var binding: ActivityDetailBinding
    private var uuidString: String? = null
    private var profileImageBase64: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        initializeViews()
        getData()
        setupEditButton()
        setupCallButton()
        setupMessageButton()
        displayProfileImage()
    }

    private fun initializeViews() {
        binding = ActivityDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    private fun getData() {
        val intent = intent
        uuidString = intent.getStringExtra("uuid")
        binding.tvName.text = intent.getStringExtra("name")
        binding.tvPhoneValue.text = intent.getStringExtra("phone")
        binding.tvAddressValue.text = intent.getStringExtra("address")
        profileImageBase64 = intent.getStringExtra("profileImage")
    }

    private fun setupEditButton() {
        val editButton: FloatingActionButton = findViewById(R.id.btn_edit)
        editButton.setOnClickListener {
            val intent = Intent(this@DetailActivity, EditUserActivity::class.java)
            intent.putExtra("id", uuidString)
            intent.putExtra("name", binding.tvName.text.toString())
            intent.putExtra("phone", binding.tvPhoneValue.text.toString())
            intent.putExtra("address", binding.tvAddressValue.text.toString())
            intent.putExtra("profileImage", profileImageBase64)
            startActivity(intent)
        }
    }

    private fun displayProfileImage() {
        if (!profileImageBase64.isNullOrEmpty()) {
            try {
                val decodedBytes = Base64.decode(profileImageBase64, Base64.DEFAULT)
                val bitmap = BitmapFactory.decodeByteArray(decodedBytes, 0, decodedBytes.size)
                binding.ivProfile.setImageBitmap(bitmap)
            } catch (e: IllegalArgumentException) {
                Log.e("DetailActivity", "Base64 디코딩 실패: ${e.message}")
            }
        }
    }

    private fun setupCallButton() {
        val phoneNumber = { binding.tvPhoneValue.text.toString() }
        findViewById<com.google.android.material.button.MaterialButton>(R.id.btn_call).setOnClickListener {
            val callIntent = Intent(Intent.ACTION_DIAL)
            callIntent.data = Uri.parse("tel:" + phoneNumber())
            startActivity(callIntent)
        }
    }

    private fun setupMessageButton() {
        val phoneNumber = { binding.tvPhoneValue.text.toString() }
        findViewById<com.google.android.material.button.MaterialButton>(R.id.btn_message).setOnClickListener {
            val messageIntent = Intent(Intent.ACTION_SENDTO)
            messageIntent.data = Uri.parse("smsto:" + phoneNumber())
            startActivity(messageIntent)
        }
    }
} 